Feature/next 9

rolivegab opened the pull request:
<!-- IMPORTANTE: Por favor confira o arquivo CONTRIBUTING.md para ver o guia de contribuição detalhado e remova os itens que não estiver usando. -->

## Contexto
<!-- Qual problema está tentando resolver? -->

Esse PR atualiza a versão do next 9 do bonde-public, além de fazer as conversões e atualizações necessárias que vêm junto com a dependência.

As configurações de `publicRuntimeConfig` para inserção das variáveis de ambiente não funcionam com o modo de build do tipo *serverless*, portanto, tive que alterar para utilizar o plugin dotenv-webpack, que substitui as variáveis process.env pelo seu valor estático descrito no arquivo `.env` do bonde-public.

Outra coisa, é que com a atualização das dependências do next-redux-wrapper, a forma de integração com o store muda, e passa a ser configurada diretamente no arquivo `_app.js`.

Também retirei de dentro da pasta `pages` todos os scripts que não são tratados como uma página que deve ser renderizada. Isso impede que o usuário acesse diretamente, por exemplo, páginas como http://www.190-mrec-detector-de-mentiras.staging.bonde.org/mobilization.connected, as quais obviamente mesmo que não renderizem nada de fato, ocasionam um erro interno no servidor 

Próximo dois passos serão: buildar com o target `serverless`, conversão de todos os scrips para o typescript.

## Issues linkadas
- [ ] #1174 
<!-- Adicione as respectivas issues linkadas a este PR. -->

## Como testar?
<!-- Adicione algumas instruções de como os reviewers podem testar esse PR. -->
Basta acessar qualquer link público de qualquer comunidade, e verificar se tudo está funcionando de acordo.

**Novas dependências**:
```
"dotenv-webpack": "^1.7.0",
"next-redux-wrapper": "^3.0.0-alpha.3",
"react": "16.8.6",
"next": "^9.0.2",
"react-dom": "16.8.6",
```

rolivegab committed:
feat(public): change process.env method

rolivegab committed:
feat(public): create _app.js

rolivegab committed:
feat(public): change redux structure

rolivegab committed:
chore(public): update dependencies

rolivegab committed:
chore(public): move root scripts to src

rolivegab committed:
chore(public): move non-pages out of pages folder

lpirola commented:
@rolivegab encontrei esse passo a passo que pode ajudar nesse processo de entendermos como subir uma função capaz de integrar com o serverless do nextejs. Cria uma issue para a outra parte desse trabalho e continuamos a discussão de lá.

https://serverless.com/framework/docs/providers/kubeless/guide/quick-start/

rolivegab committed:
chore(public): add .editorconfig

rolivegab committed:
feat(public): add global env types

rolivegab committed:
feat(public): next.js typescript integration

rolivegab committed:
chore(public): update deps

- apollo-cache-inmemory
- apollo-client
- apollo-link-http
- graphql
- graphql/tag
- react-apollo
- @types/graphql
- @types/node
- @types/react
- typescript

rolivegab committed:
feat(public): create _app.js default file

rolivegab committed:
feat(public): typescript integration

rolivegab committed:
feat(public): new apollo-client integration

alalbux reviewed (approved):
tá top, só algumas coisas que o codacy reclamou https://app.codacy.com/app/igr-santos/bonde-client/pullRequest?prid=3872794
